std/file.d

```diff
@@ -2318,7 +2318,9 @@ private bool ensureDirExists(in char[] pathname)
  * Throws: $(D FileException) on error.
  */
 
-void mkdirRecurse(in char[] pathname)
+void mkdirRecurse(R)(R pathname)
+if (isInputRange!R && !isInfinite!R && isSomeChar!(ElementEncodingType!R) &&
+    !isConvertibleToString!R)
```

Comment:
`mkdirRecurse` accepts an arbitrary range then tries to call `ensureDirExists` which accepts only `in char[]`. `ensureDirExists` thus either needs to be changed to accept a range or `mkdirRecurse`'s signature change undone.